code-studio/src/js/components/progress/progress_dot.jsx

```diff
@@ -156,49 +162,73 @@ export const ProgressDot = React.createClass({
+    const isLockedReview = isPeerReview && level.status === 'not_tried';
 
-    return (
-      <a
-        key='link'
-        href={level.url}
-        onClick={this.props.saveAnswersBeforeNavigation && dotClicked.bind(null, level.url)}
-        style={[styles.outer, showLevelName && {display: 'table-row'}]}
-      >
-        {level.icon ?
-          <i
-            className={`fa ${level.icon}`}
-            style={[
+    const iconElement = (level.icon && !isPeerReview) ?
+      <i
+        className={`fa ${level.icon}`}
+        style={[
+              styles.dot.common,
               styles.dot.puzzle,
               this.props.courseOverviewPage && styles.dot.overview,
               styles.dot.icon,
               smallDot && styles.dot.icon_small,
               level.status && level.status !== 'not_tried' && styles.dot.icon_complete,
               outlineCurrent && {textShadow: createOutline(color.level_current)}
             ]}
-          /> :
-          <div
-            className={`level-${level.id}`}
-            style={[
-              styles.dot.puzzle,
+        /> :
+      <div
+        className={`level-${level.id} ${isPeerReview && `fa ${level.icon}`}`}
+        style={[
+              styles.dot.common,
+              isLockedReview ? styles.dot.lockedReview : styles.dot.puzzle,
               this.props.courseOverviewPage && styles.dot.overview,
               smallDot && styles.dot.small,
               level.kind === 'assessment' && styles.dot.assessment,
               outlineCurrent && {borderColor: color.level_current},
               showUnplugged && styles.dot.unplugged,
-              styles.status[level.status || 'not_tried']
+              styles.status[level.status || 'not_tried'],
             ]}
-          >
-            {level.kind === 'named_level' ? '\u00a0' : level.title}
-          </div>
-        }
-        {showLevelName &&
-          <span key='named_level' style={styles.levelName}>{level.name}</span>
-        }
-      </a>
-    );
+        >
+        {(showLevelName || (isPeerReview && level.icon === '')) ? '\u00a0' : level.title}
+      </div>;
+
+    const rowText = (showLevelName || isPeerReview) &&
+      <span
+        key='named_level'
+        style={[styles.levelName, isLockedReview && {color: color.charcoal}]}
+      >
+        {level.name}
+      </span>;
+
+
+    if (isLockedReview) {
+      return (
+        <div
+          key='lockedReview'
+          style={[styles.outer, {display: 'table-row'}]}
+        >
+          {iconElement}
+          {rowText}
+        </div>
+      );
+    } else {
+      return (
+        <a
+          key='link'
+          href={level.url}
+          onClick={this.props.saveAnswersBeforeNavigation && dotClicked.bind(null, level.url)}
+          style={[styles.outer, (showLevelName || isPeerReview) && {display: 'table-row'}]}
+        >
+          {iconElement}
+          {rowText}
+        </a>
+      );
+    }
```

Comment:
[Same feedback as before](https://github.com/code-dot-org/code-dot-org/pull/9221#discussion_r68854287): prefer putting all the JSX inside the return (...) of render() rather than using local variables.